app/scripts/components/common/mapbox/layers/utils.ts

```diff
@@ -258,3 +262,95 @@ export function resolveLayerTemporalExtent(
     `Invalid time density [${timeDensity}] on layer [${layerData.id}]`
   );
 }
+
+// There are cases when the data can't be displayed properly on low zoom levels.
+// In these cases instead of displaying the raster tiles, we display markers to
+// indicate whether or not there is data in a given location. When the user
+// crosses the marker threshold, if below the min zoom we have to request the
+// marker position, and if above we have to register a mosaic query. Since this
+// switching can happen several times, we cache the api response using the
+// request params as key.
+const quickCache = new Map<string, any>();
+export async function requestQuickCache(
+  url: string,
+  payload,
+  controller: AbortController
+) {
+  const key = `${url}${JSON.stringify(payload)}`;
+
+  // No cache found, make request.
+  if (!quickCache.has(key)) {
+    const response = await axios.post(url, payload, {
+      signal: controller.signal
+    });
+    quickCache.set(key, response.data);
+  }
+  return quickCache.get(key);
+}
+
+/**
+ * Creates the appropriate filter object to send to STAC.
+ *
+ * @param {Date} date Date to request
+ * @param {string} collection STAC collection to request
+ * @returns Object
+ */
+export function getFilterPayload(date: Date, collection: string) {
+  return {
+    op: 'and',
+    args: [
+      {
+        op: '>=',
+        args: [{ property: 'datetime' }, userTzDate2utcString(startOfDay(date))]
+      },
+      {
+        op: '<=',
+        args: [{ property: 'datetime' }, userTzDate2utcString(endOfDay(date))]
+      },
+      {
+        op: 'eq',
+        args: [{ property: 'collection' }, collection]
+      }
+    ]
+  };
+}
+
+export function getMergedBBox(features: StacFeature[]) {
+  const mergedBBox = [
+    Number.POSITIVE_INFINITY,
+    Number.POSITIVE_INFINITY,
+    Number.NEGATIVE_INFINITY,
+    Number.NEGATIVE_INFINITY
+  ] as [number, number, number, number];
+  features.forEach((feature) => {
+    if (feature.bbox[0] < mergedBBox[0]) mergedBBox[0] = feature.bbox[0];
+    if (feature.bbox[1] < mergedBBox[1]) mergedBBox[1] = feature.bbox[1];
+    if (feature.bbox[2] > mergedBBox[2]) mergedBBox[2] = feature.bbox[2];
+    if (feature.bbox[3] > mergedBBox[3]) mergedBBox[3] = feature.bbox[3];
+  });
+  return mergedBBox;
```

Comment:
I'm often torn between a more imperative approach, and one where you exclusively use high-order function without any sort of variable reassignement.

Modifying a variable makes code easier to read, but forces you to keep track of where a variable is modified (especially tricky with `const` arrays and objects where you modify the contents.)

The same approach without var reassignment would be:
```js
return features.reduce(
    (acc, feature) =>
      acc.map((mergeCoord, idx) => {
        const featCoord = feature.bbox[idx];

        if (idx < 2) {
          // SW
          return featCoord < mergeCoord ? featCoord : mergeCoord;
        } else {
          // NE
          return featCoord > mergeCoord ? featCoord : mergeCoord;
        }
      }),
    mergedBBox
  );
```

- However, isn't this approach harder to reason about?
- Should it be argued that modifying variables is ok for a simple function but a behavior to be avoided in general?
- How do we draw the line?
- Is there a happy middle ground? Like:
```js
return features.reduce(
    (acc, feature) => [
      feature.bbox[0] < acc[0] ? feature.bbox[0] : acc[0],
      feature.bbox[1] < acc[1] ? feature.bbox[1] : acc[1],
      feature.bbox[2] > acc[2] ? feature.bbox[2] : acc[2],
      feature.bbox[3] > acc[3] ? feature.bbox[3] : acc[3]
    ],
    mergedBBox
  );
```

@nerik Would like your thoughts.
cc @hanbyul-here

Reply:
my 💭 : Not changing the variable is a generally great rule of thumb (and it often feels more exciting? to write in a declarative way). However, considering iteration methods still have different performances, we might need to be flexible on the rule. As far as I checked last time, `foreach` still has better performances than declarative methods such as `reduce`?

Reply:
@danielfdsilva @hanbyul-here Thank you! I adopted the "happy middle ground approach". 
- using `reduce` over `forEach` does feel more elegant
- the first approach is just too difficult to read IMHO
  - thinking of me re-reading that in 6 months
  - we have 4 items, with the need for logical branching for half of them, so in reality you are abstracting away logic for 2 items. Worth it?
  - or put more simply: "simplifying" the code by writing more lines of code: probably not needed and/or premature 
- re performance: I believe where you really see a difference is between `for` loops and all other declarative approaches (ie `reduce`, `map`, `forEach`, etc), but even though it doesn't really matter in this order of magnitude of iterations